### DISCUSSION TITLE
Compiling as a xcframework

### DISCUSSION THREAD

**Discussion by cala4** [pointfreeco/swift-composable-architecture#890] [Q&A]
Has anyone been successful in compiling TCA into a xcframework along with all its dependencies?

I've managed to compile everything except the dependency OrderedCollections which, after importing it, doesn't seem to work.

If anyone has had any success in integrating TCA into their projects using xcframeworks instead of SPM please share how you were able to achieve it. Been banging my head with this for quite a few hours already :)

Thanks!

**Comment by stonecrumbler** [comment]
If anyone interested I found the solution that seems to work fine:
```swift 
@_implementationOnly import ComposableArchitecture
```

**Comment by flockoffiles** [comment]
I would also be interested to know the way to compile TCA as an xcframework. The straightforward approach does not work for me for some reason - at least I am not getting .swiftinterface files included.
Perhaps, adding a scheme (to build TCA as an xcframework) to the standard Xcode Workspace would solve the problem for many people all at once...

Here is an example command I use to build the Simulator slice:
xcrun xcodebuild archive -workspace "ComposableArchitecture.xcworkspace" -scheme "ComposableArchitecture" -configuration "Debug" -destination "generic/platform=iOS Simulator" -archivePath "MY_ARCHIVE_PATH.xcarchive" -sdk "iphonesimulator" BUILD_LIBRARY_FOR_DISTRIBUTION=YES SKIP_INSTALL=NO CLANG_ENABLE_CODE_COVERAGE=NO  GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO SWIFT_SERIALIZE_DEBUGGING_OPTIONS=NO OTHER_SWIFT_FLAGS="-Xfrontend -no-serialize-debugging-options" DEBUG_INFORMATION_FORMAT=dwarf-with-dsym

(And the similarly for device)

I also explicitly set the library type to .dynamic in Package.swift

**Reply by flockoffiles** [reply]
I created a issue for OrderedCollections here: https://github.com/apple/swift-collections/issues/157

**Reply by flockoffiles** [reply]
I have discovered that if I simply comment out the typealias defined in the UnorderedView
OrderedSet+UnorderedView.swift, line 17

and then fix the line where it is used:

OrderedSet+UnorderedView.swift, line 305

Then the resulting .swiftinterface file compiles again.

Note: I tried it with Xcode 13.3 and 14 beta 5.
This seems like a compiler limitation to me.

**Comment by flockoffiles** [comment]
Apparently, the problem is now fixed for OrderedCollections (see https://github.com/apple/swift-collections/issues/157). The fix will become available with their next release.
Maybe, @stephencelis could include the updated dependency when it's available?

**Reply by miguelkennedy** [reply]
@flockoffiles 

Is there any chance you can share the entire shell script you're using to compile the Composable Architecture to a xcframework? I'm facing some issues :/

Do you have to compile all dependencies one by one into individual xcframeworks as well?

Thanks in advance!

**Reply by flockoffiles** [reply]
I used an open source tool (create-xcframework) to do this.

You need to call it on all of the dependencies of composable architecture as well (to generate xcframeworks for them).

Example:
```bash
swift create-xcframework OrderedCollections --configuration debug --stack-evolution --platform ios --xc-setting CLANG_ENABLE_CODE_COVERAGE=NO --xc-setting GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO --xc-setting SWIFT_SERIALIZE_DEBUGGING_OPTIONS=NO --xc-setting OTHER_SWIFT_FLAGS="-Xfrontend -no-serialize-debugging-options" --xc-setting DEBUG_INFORMATION_FORMAT=dwarf-with-dsym
```

**Reply by miguelkennedy** [reply]
Thank you! Will give it a try later today. Much appreciated 👍🏼
